### ISSUE TITLE
CA1831 and CA1846 don't fire when combining range syntax with `Substring`

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn-analyzers#6907]
### Analyzer

**Diagnostic ID**:
[CA1831](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1831): `Use AsSpan instead of Range-based indexers for string when appropriate`
[CA1846](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1846): `Prefer AsSpan over Substring`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 8.0.100-preview.7.23376.3

### Describe the bug

The following code generates [CA1831](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1831): `Use AsSpan instead of Range-based indexers for string when appropriate`:

```c#
void Foo(string str, int position)
{
    var foo = string.Concat(str[0..position], str.AsSpan(position + 1));
}
```
And the following code generates [CA1846](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1846): `Prefer AsSpan over Substring`:

```c#
void Foo(string str, int position)
{
    var foo = string.Concat(str.Substring(0, position), str.Substring(position + 1));
}
```
All is good. In both cases, an analyzer warning is issued suggesting more efficient code. However, if I combine these two - use the range syntax in one argument and `Substring` in the other, like this:
```c#
void Foo(string str, int position)
{
    var foo = string.Concat(str[0..position], str.Substring(position + 1));
}
```
Suddenly, no diagnostic is issued. Both analyzers are silent. I think this code should produce a warning as well, be it either CA1831 or CA1846, or both.

**Comment by mavasani** [comment]
@buyaa-n
